.github/workflows/Compile-and-run.yml

```diff
@@ -0,0 +1,25 @@
+# IMPORTANT NOTE: The results of these benchmarks are not representative. This
+# is simply a smoke test to check everything compiles and runs.
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  Compile-and-run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-22.04, macos-14, windows-2022 ]
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - name: Run build and benchmark script
+        run: python Benchmark/run-benchmarks.py -- -G"Unix Makefiles"
```

Comment:
I know that GH Runner are not a good platform to tests this, but it is so nice to see the benchmarks there... How hard would it be to split this step in "build" and "benchmark"?

Reply:
Done! Let me know what you think :-)

Reply:
I think it was better before, sorry :sweat:

Reply:
Reverted